credentials/java/com/google/auth/Credentials.java

```diff
@@ -21,13 +22,25 @@
   /**
    * Get the current request metadata, refreshing tokens if required.
    *
+   * <p>For temporary compatibility only. Passes a null URI.
+   *
+   * @throws IOException if there was an error getting up-to-date access.
+   */
+  public Map<String, List<String>> getRequestMetadata() throws IOException {
+    return getRequestMetadata(null);
+  }
+
+  /**
+   * Get the current request metadata, refreshing tokens if required.
+   *
    * <p>This should be called by the transport layer on each request, and the data should be
    * populated in headers or other context. The operation can block and fail to complete and may do
    * things such as refreshing access tokens.
    *
+   * @param uri URI of the entry point for the request.
    * @throws IOException if there was an error getting up-to-date access.
    */
-  public abstract Map<String, List<String>> getRequestMetadata() throws IOException;
+  public abstract Map<String, List<String>> getRequestMetadata(URI uri) throws IOException;
```

Comment:
Can the URI be provided at Credential construction time? From talking to Jose the cardinality of 'audience' values is likely to be quite low (just as for scopes) so it may be more practical to have a credential-instance-per-audience than requiring it to be provided per-call by some framework layer below the application code. e.g. the ClientAuthInterceptor in grpc.

@casiboy